nano/node/bootstrap.cpp

```diff
@@ -3161,3 +3170,67 @@ void nano::frontier_req_server::next ()
 	current = account_pair.first;
 	frontier = account_pair.second;
 }
+
+void nano::pulls_cache::add (nano::pull_info const & pull_a)
+{
+	if (pull_a.processed > 500)
+	{
+		std::lock_guard<std::mutex> guard (pulls_cache_mutex);
+		// Clean old pull
+		if (cache.size () >= max_cache)
+		{
+			cache.erase (cache.begin ());
```

Comment:
Is it possible for `cache.size ()` to ever be greater than `max_cache`? Perhaps that should be an assert (programmer error). If it can (perhaps I'm missing something) then it should erase a range of elements so that `cache.size` always becomes less than `max_cache`, rather than just erasing one.

Reply:
Adding assert, thanks